### PR TITLE
feat: add an option to enable DirectPath xDS

### PR DESCRIPTION
**Pull request by mohanli-ml**
Currently DirectPath xDS is enabled by using env. We want to also provide an option for this.

Java PR: https://github.com/googleapis/gax-java/pull/1968

@veblush

### CODE REVIEW

**Comment by mohanli-ml** [comment]
@codyoss Hi Cody, can you help to review PR? Thanks!

**Review by codyoss — APPROVED** [review]

File: option/internaloption/internaloption.go [hunk 1]
```diff
@@ -67,6 +67,21 @@ func (e enableDirectPath) Apply(o *internal.DialSettings) {
 	o.EnableDirectPath = bool(e)
 }
 
+// UseDirectPathXds returns a ClientOption that overrides the default
+// DirectPath type. It is only valid when DirectPath is enabled.
+//
+// It should only be used internally by generated clients.
+// This is an EXPERIMENTAL API and may be changed or removed in the future.
+func UseDirectPathXds(xds bool) option.ClientOption {
```
Comment (codyoss) [inline option/internaloption/internaloption.go:75]: nit: maybe just make this option take not args and be called EnableDirctPathXds? No need to pass a bool, the presence of the option is good enough

Comment (noahdietz) [reply option/internaloption/internaloption.go:75]: +1

Comment (mohanli-ml) [reply option/internaloption/internaloption.go]: Done. Thanks for reviewing!

**Pull request merged**